src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Completion/DirectiveAttributeCompletionItemProviderBase.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+using RazorSyntaxList = Microsoft.AspNetCore.Razor.Language.Syntax.SyntaxList<Microsoft.AspNetCore.Razor.Language.Syntax.SyntaxNode>;
+using RazorSyntaxNode = Microsoft.AspNetCore.Razor.Language.Syntax.SyntaxNode;
+
+namespace Microsoft.CodeAnalysis.Razor.Completion
+{
+    internal abstract class DirectiveAttributeCompletionItemProviderBase : RazorCompletionItemProvider
```

Comment:
Could have been a set of static methods somewhere, choose to go with an abstract base class for no good reason 😆. These methods will also be used to implement the directive attribute parameter completion provider which is why I have them factored separately.